### PR TITLE
First pass for template generation

### PR DESCRIPTION
**Pull request by nciric**
Replacing easy part with the template. I would have to think more about the rest.
This approach would work for c/js too.

### CODE REVIEW

File: tool/Cargo.toml [hunk 1]
```diff
@@ -23,3 +23,7 @@ toml = "0.5.8"
 
 [dev-dependencies]
 insta = { version = "1.7.1", features = [ "backtrace" ] }
+
+[dependencies.tera]
```
Comment (Manishearth) [inline tool/Cargo.toml:27]: we should use a normal `tera = { version = "1", default-features = false }` in the `[dependencies]` section instead of this

**Review by Manishearth — APPROVED** [review]
Overall looks good, excited to see more work along these lines!

**Comment by CBenoit** [comment]
Happy to see this happening too! :tada:

File: tool/Cargo.toml (hunk 1 above)
Comment (nciric) [reply tool/Cargo.toml]: Fixed.

**Review by Manishearth — APPROVED** [review]

**Pull request merged**